Sources/Jobs/Scheduler/ScheduleBuilder.swift

```diff
@@ -299,6 +299,55 @@ public final class ScheduleBuilder {
     var dayOfWeek: DayOfWeek?
     var time: Time?
     var minute: Minute?
+
+    /// returns the next date that satisfied the created schedule
+    internal func resolveNextDateThatSatisifiesSchedule(date: Date) throws -> Date {
+
+        var monthConstraint: MonthRecurrenceRuleConstraint? = nil
+        if let monthValue = month?.rawValue {
+            monthConstraint = try MonthRecurrenceRuleConstraint.atMonth(monthValue)
+        }
+
+        var dayOfMonthConstraint: DayOfMonthRecurrenceRuleConstraint? = nil
+        if let dayValue = day {
+            switch dayValue {
+            case .first:
+                dayOfMonthConstraint = try DayOfMonthRecurrenceRuleConstraint.atDayOfMonth(1)
+            case .last:
+                dayOfMonthConstraint = try DayOfMonthRecurrenceRuleConstraint.atDaysOfMonth([28, 29, 30, 31])
```

Comment:
I'm not sure I understand why we pass in all four values here - is that because the month will only resolve whichever of these is the actual last value? I'm thinking of July, for example, which has valid dates at all of these.

Reply:
Good catch on this. It should be calling `DayOfMonthRecurrenceRuleConstraint.atLastDayOfMonth()`. Which sets the `isLimitedToLastDayOfMonth` property to true on `DayOfMonthRecurrenceRuleConstraint`. In the next PR you will see nextValidValue is essentially overwritten to only return the last day of the month.

Reply:
Awesome! Thanks for clarifying